tests/functional/afr/test_self_heal_with_full_algorithm.py

```diff
@@ -0,0 +1,207 @@
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+"""
+Description:
+    Test self heal when data-self-heal-algorithm option is set to full.
+"""
+
+from random import sample
+
+from glusto.core import Glusto as g
+from glustolibs.gluster.brick_libs import (bring_bricks_offline,
+                                           are_bricks_offline)
+from glustolibs.gluster.exceptions import ExecutionError
+from glustolibs.gluster.gluster_base_class import GlusterBaseClass, runs_on
+from glustolibs.gluster.heal_libs import (is_volume_in_split_brain,
+                                          monitor_heal_completion)
+from glustolibs.gluster.lib_utils import collect_bricks_arequal
+from glustolibs.gluster.volume_ops import (volume_start,
+                                           set_volume_options)
+from glustolibs.gluster.volume_libs import (
+    verify_all_process_of_volume_are_online, get_subvols)
+
+
+@runs_on([['arbiter', 'distributed-arbiter', 'replicated',
+           'distributed-replicated'], ['glusterfs']])
+class TestSelfHealWithFullAlgorithm(GlusterBaseClass):
+    def setUp(self):
+
+        # Calling GlusterBaseClass setUp
+        self.get_super_method(self, 'setUp')()
+
+        if not self.setup_volume_and_mount_volume(self.mounts):
+            raise ExecutionError("Unable to setup and mount volume")
+        g.log.info("Volume created and mounted successfully")
+
+    def tearDown(self):
+
+        # Unmount and cleanup the volume
+        if not self.unmount_volume_and_cleanup_volume(self.mounts):
+            raise ExecutionError("Unable to unmount and cleanup volume")
+        g.log.info("Volume unmounted and deleted successfully")
+
+        # Calling GlusterBaseClass Teardown
+        self.get_super_method(self, 'tearDown')()
+
+    def test_self_heal_with_full_algorithm(self):
+        """
+        Test Steps:
+        1. Create a replicated/distributed-replicate volume and mount it
+        2. Set data/metadata/entry-self-heal to off and
+           data-self-heal-algorithm to full
+        3. Create a directory from the mount point
+        4. Bring down all bricks from a selected set
+        5. Create few files inside the directory with some data
+        6. Bring back the bricks online and wait for heal to complete
+        7. Check arequal of the subvol and all the bricks in the subvol should
+           have same checksum
+        8. Bring down all bricks from another set
+        9. Modify the data of existing files under the directory
+        10. Bring back the bricks online and wait for heal to complete
+        11. Check arequal of the subvol and all the brick in the same subvol
+            should have same checksum
+        """
+
+        # Setting options
+        for key, value in (("data-self-heal", "off"),
```

Comment:
As you already know client heal ops are set to `off` by default. Do you still want to set these three options explicitly?

Reply:
Setting this explicitly will make it future proof.